coordinator/src/dlc_protocol.rs

```diff
@@ -222,6 +223,50 @@ impl DlcProtocolExecutor {
         Ok(())
     }
 
+    /// Finishes a dlc protocol by the corresponding dlc protocol type handling.
+    pub fn finish_dlc_protocol(
+        &self,
+        protocol_id: ProtocolId,
+        trader_id: &PublicKey,
+        contract_id: &ContractId,
+        channel_id: &DlcChannelId,
+    ) -> Result<()> {
+        let mut conn = self.pool.get()?;
+        conn.transaction(|conn| {
+            let dlc_protocol = db::dlc_protocols::get_dlc_protocol(conn, protocol_id)?;
+
+            match dlc_protocol.protocol_type {
+                DlcProtocolType::Open { trade_params }
+                | DlcProtocolType::Renew { trade_params } => self.finish_trade_dlc_protocol(
+                    conn,
+                    trade_params,
+                    protocol_id,
+                    false,
```

Comment:
It's still the same argument to keep it as with the refactoring there shouldn't be a different handling for closing or opening a position. :)

However, since you mentioned it now the second time, I will rewrite it :)